plugins/standard_tests/newhelm/tests/safe.py

```diff
@@ -0,0 +1,115 @@
+import csv
+from pydantic import BaseModel
+from typing import List, Dict, Mapping
+
+from newhelm.annotators.llama_guard_annotator import (
+    LlamaGuardAnnotation,
+    LlamaGuardAnnotator,
+)
+from newhelm.aggregations import mean_of_measurement
+from newhelm.base_annotator import BaseAnnotator
+from newhelm.base_test import BasePromptResponseTest, Result, TestMetadata
+from newhelm.dependency_helper import DependencyHelper
+from newhelm.external_data import ExternalData, WebData
+from newhelm.prompt import TextPrompt, SUTOptions
+from newhelm.record_init import record_init
+from newhelm.single_turn_prompt_response import (
+    MeasuredTestItem,
+    PromptWithContext,
+    TestItemAnnotations,
+    TestItem,
+)
+from newhelm.test_registry import TESTS
+
+
+HAZARDS_MAPPING = {
+    "cae": "Child abuse and exploitation",
+    "ssh": "Suicide, self-harm and eating disorders",
+    "ter": "Terrorism",
+}
+
+
+class SAFETestItemContext(BaseModel):
+    prompt_uid: str
+    hazard: str
+
+
+class SAFE(BasePromptResponseTest):
+    @record_init
+    def __init__(self, hazard: str, data_source: str):
+        assert hazard in HAZARDS_MAPPING, f"Invalid hazard type {hazard}."
+        self.hazard = hazard
+        self.hazard_name = HAZARDS_MAPPING[hazard]
+        self.data_source = data_source
+
+    def get_metadata(self) -> TestMetadata:
+        return TestMetadata(
+            name=f"SAFE-{self.hazard}",
+            description=f"Safe AI with Flexible Evaluation for {self.hazard_name}",
+        )
+
+    def get_dependencies(self) -> Mapping[str, ExternalData]:
+        return {
+            "prompts_csv": WebData(
+                source_url=f"{self.data_source}/export?format=csv",
+            ),
+        }
+
+    def make_test_items(self, dependency_helper: DependencyHelper) -> List[TestItem]:
+        sut_options = SUTOptions(
+            max_tokens=500,
+            temperature=0.01,
+        )
+        test_items: List[TestItem] = []
+        data_file = dependency_helper.get_local_path("prompts_csv")
+        with open(data_file, "r") as csvfile:
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                assert row["Hazard type"] == self.hazard_name
+                text = row["Direct prompt"].strip()
+                prompt = PromptWithContext(
+                    prompt=TextPrompt(text=text, options=sut_options),
+                )
+                context = SAFETestItemContext(
+                    prompt_uid=row["UID"],
+                    hazard=row["Hazard type"],
```

Comment:
I believe all prompts for this instance of the Test will have the same `hazard`, which is the value of `self.hazard_name`. Furthermore nothing downstream seems to care about this value. Can we remove it from the context?

If so, can we remove context entirely? As I said on #215 I don't think UID makes sense in context.

Reply:
I can remove the hazard, but the prompt id is useful if someone wants to do a closer inspection of how a model responded to particular prompt. And it was requested by the working group. Do you think we can keep it here for now until we tackle the prompt UID question in newhelm at large?

Reply:
Removed entirely.